presentation/common/src/presentation-common/PropertyFormatter.ts

```diff
@@ -79,17 +101,119 @@ export class ContentPropertyValueFormatter extends PropertyValueFormatter {
   private isFormattable(field: Field): field is FormattableField {
     return field.isPropertiesField()
       && field.properties.length > 0
-      && field.properties[0].property.kindOfQuantity !== undefined
-      && field.properties[0].property.kindOfQuantity.activeFormat !== undefined;
+      && field.properties[0].property.kindOfQuantity !== undefined;
   }
 }
 
 type FormattableField = PropertiesField & {
   properties: [{
     property: PropertyInfo & {
-      kindOfQuantity: KindOfQuantityInfo & {
-        activeFormat: FormatProps;
-      };
+      kindOfQuantity: KindOfQuantityInfo;
     };
   }];
 };
+
+interface FormattingProps {
+  formatProps: FormatProps;
+  persistenceUnitName: string;
+}
+
+async function getFormattingProps(schemaLocater: ISchemaLocater, options: FormatOptions): Promise<FormattingProps | undefined> {
+  const { koqName, unitSystem } = options;
+
+  const koq = await getKoq(schemaLocater, koqName);
+  if (!koq)
+    return undefined;
+
+  const persistenceUnit = await koq.persistenceUnit;
+  // istanbul ignore if
+  if (!persistenceUnit)
+    return undefined;
+
+  const format = await getKoqFormat(koq, unitSystem);
+  if (!format)
+    return undefined;
+
+  return { formatProps: formatToFormatProps(format), persistenceUnitName: persistenceUnit.fullName };
+}
+
+async function getKoq(schemaLocater: ISchemaLocater, fullName: string) {
+  const [schemaName, propKoqName] = fullName.split(":");
+  const schema = await schemaLocater.getSchema(new SchemaKey(schemaName), SchemaMatchType.Latest);
+  if (!schema)
+    return undefined;
+
+  return schema.getItem<KindOfQuantity>(propKoqName);
+}
+
+async function getKoqFormat(koq: KindOfQuantity, unitSystem: UnitSystemKey) {
+  const unitSystems = getUnitSystemGroupNames(unitSystem);
+  const presentationFormat = await getKoqPresentationFormat(koq, unitSystems);
+  if (presentationFormat)
+    return presentationFormat;
+
+  return koq.defaultPresentationFormat;
```

Comment:
should attempt to use persistence unit if it's in requested unit system and only then fall back to default presentation format

Reply:
Added format for persistence unit